packages/eas-cli/src/build/graphql.ts

```diff
@@ -42,6 +42,7 @@ export function transformProjectArchive(archiveSource: ArchiveSource): ProjectAr
 export function transformMetadata(metadata: Metadata): BuildMetadataInput {
   return {
     ...metadata,
+    fingerprintSource: metadata.fingerprintSource as BuildMetadataInput['fingerprintSource'],
```

Comment:
Can we do `assert(metadata.fingerprintSource.type === GCS)` instead of type casting here?

Reply:
Doesn't work
<img width="917" alt="Screenshot 2024-06-28 at 10 46 22" src="https://github.com/expo/eas-cli/assets/6534400/4d2edde3-dc07-404f-b66b-ed257942820d">

If I use the enum from gql it also doesn't work
<img width="962" alt="Screenshot 2024-06-28 at 10 47 54" src="https://github.com/expo/eas-cli/assets/6534400/66d42c20-1083-484a-85a9-277423020451">

This is the gql one:
```ts
export enum FingerprintSourceType {
  Gcs = 'GCS'
}
```

This is the `@expo/eas-build-job` one:
```ts
export declare enum FingerprintSourceType {
    'GCS' = "GCS",
    'PATH' = "PATH",
    'URL' = "URL"
}
```

I couldn't figure out how to make it work anyway.